src/Cake.Common/Xml/XmlPokeAliases.cs

```diff
@@ -0,0 +1,177 @@
+        /// <param name="value">The value to set too. Leave blank to remove the selected nodes.</param>
+        /// <param name="settings">Additional settings to tweak Xml Poke behavior.</param>
+        [CakeMethodAlias]
+        public static void XmlPoke(this ICakeContext context, FilePath filePath, string xpath, string value, XmlPokeSettings settings = null)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            if (filePath == null)
+            {
+                throw new ArgumentNullException("filePath");
+            }
+
+            if (string.IsNullOrWhiteSpace(xpath))
+            {
+                throw new ArgumentNullException("xpath");
+            }
+
+            if (settings == null)
+            {
+                settings = new XmlPokeSettings();
+            }
+
+            var sourceFile = context.FileSystem.GetFile(filePath);
+            if (!sourceFile.Exists)
+            {
+                throw new FileNotFoundException("Source File not found.", sourceFile.Path.FullPath);
+            }
+            
+            using (var sourceReader = sourceFile.Open(FileMode.Open, FileAccess.ReadWrite, FileShare.None))
+            {
+                var xmlReader = XmlReader.Create(sourceReader);
+                var xmlWriter = XmlWriter.Create(sourceReader);
+                XmlPoke(context, xmlReader, xmlWriter, xpath, value, settings);
+            }
+        }
+
+        /// <summary>
+        /// Set the value of, or remove, target nodes.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="sourceXml">The source xml to transform.</param>
+        /// <param name="xpath">The xpath of the nodes to set.</param>
+        /// <param name="value">The value to set too. Leave blank to remove the selected nodes.</param>
+        /// <param name="settings">Additional settings to tweak Xml Poke behavior.</param>
+        /// <returns>Resulting XML.</returns>
+        [CakeMethodAlias]
+        public static string XmlPoke(this ICakeContext context, string sourceXml, string xpath, string value, XmlPokeSettings settings = null)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            if (string.IsNullOrWhiteSpace(sourceXml))
+            {
+                throw new ArgumentNullException("sourceXml");
+            }
+
+            if (string.IsNullOrWhiteSpace(xpath))
+            {
+                throw new ArgumentNullException("xpath");
+            }
+
+            if (settings == null)
+            {
+                settings = new XmlPokeSettings();
+            }
+
+            using (var resultStream = new MemoryStream())
+            using (var sourceReader = new StringReader(sourceXml))
+            {
+                var xmlReader = XmlReader.Create(sourceReader);
```

Comment:
I believe `XmlReader` & `XmlWriter` inherit from `IDisposable` and should be wrapped in a using statement.